src/error.rs

```diff
@@ -0,0 +1,34 @@
+use pdb_crate::Error as PDBError;
+
+
+pub type CompactSymbolResult<T> = Result<T, CompactSymbolError>;
+
+#[derive(Fail, Debug)]
+pub enum CompactSymbolError {
+    #[fail(display = "Strings do not match")]
+    UnmatchedString,
+
+    #[fail(display = "pdb_crate error: {}", _0)]
+    PDBError(PDBError),
+
+    #[fail(display = "Feature is not implemented")]
+    UnimplementedFeature,
+
+    #[fail(display = "IO Error")]
+    IOError,
+
+    #[fail(display = "Unknown error")]
+    UnknownError,
+
+    #[fail(display = "Invalid string: {}", _0)]
+    InvalidStringError(&'static str),
```

Comment:
For `InvalidStringError` and `InvalidResultError`, I collapse it to one single `InvalidInputError` which takes in a  error message parameter describing what parameter cannot be parsed specifically so to not make it generic.